src/lib/multicall/executeMulticallWorker.ts

```diff
@@ -0,0 +1,43 @@
+import { uniqueId } from "lodash";
+import { PRODUCTION_PREVIEW_KEY } from "config/localStorage";
+
+import MyWorker from "./multicall.worker";
+
+export const executorWorker: Worker = new MyWorker();
+
+executorWorker.onmessage = (event) => {
+  const { id, result, error } = event.data;
+
+  const promise = promises.get(id);
+
+  if (!promise) {
+    return;
+  }
+
+  if (error) {
+    promise.reject(error);
+  } else {
+    promise.resolve(result);
+  }
+
+  promises.delete(id);
+};
+
+const promises = new Map<string, { resolve: any; reject: any }>();
+
+export async function executeMulticallWorker(chainId: number, request: any) {
```

Comment:
let's change type to `MulticallRequestConfig<any>` for `request`

Reply:
done